From b602cd66f7addba4cd9681516679c302f2396997 Mon Sep 17 00:00:00 2001
From: Paul Cercueil <paul.cercueil@analog.com>
Date: Wed, 18 May 2016 13:02:12 +0200
Subject: [PATCH 8/8] CMake: Generate VolkConfig.cmake with a hardcoded install
 path hint

This allows the Volk headers and libraries to be properly found when
Volk was built with a prefix that differs from the expected paths.

Signed-off-by: Paul Cercueil <paul.cercueil@analog.com>
---
 CMakeLists.txt                    |  7 ++++++-
 cmake/Modules/VolkConfig.cmake    | 26 --------------------------
 cmake/Modules/VolkConfig.cmake.in | 28 ++++++++++++++++++++++++++++
 3 files changed, 34 insertions(+), 27 deletions(-)
 delete mode 100644 cmake/Modules/VolkConfig.cmake
 create mode 100644 cmake/Modules/VolkConfig.cmake.in

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b6c0656..b5f2ed0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -215,6 +215,11 @@ endif()
 ########################################################################
 
 configure_file(
+  ${CMAKE_SOURCE_DIR}/cmake/Modules/VolkConfig.cmake.in
+  ${CMAKE_BINARY_DIR}/cmake/Modules/VolkConfig.cmake
+@ONLY)
+
+configure_file(
   ${CMAKE_SOURCE_DIR}/cmake/Modules/VolkConfigVersion.cmake.in
   ${CMAKE_BINARY_DIR}/cmake/Modules/VolkConfigVersion.cmake
 @ONLY)
@@ -230,7 +235,7 @@ endif(NOT CMAKE_MODULES_DIR)
 
 install(
     FILES
-    ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules/VolkConfig.cmake
+    ${CMAKE_CURRENT_BINARY_DIR}/cmake/Modules/VolkConfig.cmake
     ${CMAKE_CURRENT_BINARY_DIR}/cmake/Modules/VolkConfigVersion.cmake
     DESTINATION ${CMAKE_MODULES_DIR}/volk
     COMPONENT "volk_devel"
diff --git a/cmake/Modules/VolkConfig.cmake b/cmake/Modules/VolkConfig.cmake
deleted file mode 100644
index 425eb01..0000000
--- a/cmake/Modules/VolkConfig.cmake
+++ /dev/null
@@ -1,26 +0,0 @@
-INCLUDE(FindPkgConfig)
-PKG_CHECK_MODULES(PC_VOLK volk)
-
-FIND_PATH(
-    VOLK_INCLUDE_DIRS
-    NAMES volk/volk.h
-    HINTS $ENV{VOLK_DIR}/include
-        ${PC_VOLK_INCLUDEDIR}
-    PATHS /usr/local/include
-          /usr/include
-)
-
-FIND_LIBRARY(
-    VOLK_LIBRARIES
-    NAMES volk
-    HINTS $ENV{VOLK_DIR}/lib
-        ${PC_VOLK_LIBDIR}
-    PATHS /usr/local/lib
-          /usr/local/lib64
-          /usr/lib
-          /usr/lib64
-)
-
-INCLUDE(FindPackageHandleStandardArgs)
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(VOLK DEFAULT_MSG VOLK_LIBRARIES VOLK_INCLUDE_DIRS)
-MARK_AS_ADVANCED(VOLK_LIBRARIES VOLK_INCLUDE_DIRS)
diff --git a/cmake/Modules/VolkConfig.cmake.in b/cmake/Modules/VolkConfig.cmake.in
new file mode 100644
index 0000000..2c2c656
--- /dev/null
+++ b/cmake/Modules/VolkConfig.cmake.in
@@ -0,0 +1,28 @@
+INCLUDE(FindPkgConfig)
+PKG_CHECK_MODULES(PC_VOLK volk)
+
+FIND_PATH(
+    VOLK_INCLUDE_DIRS
+    NAMES volk/volk.h
+    HINTS $ENV{VOLK_DIR}/include
+        ${PC_VOLK_INCLUDEDIR}
+    PATHS /usr/local/include
+          /usr/include
+          "@CMAKE_INSTALL_PREFIX@/include"
+)
+
+FIND_LIBRARY(
+    VOLK_LIBRARIES
+    NAMES volk
+    HINTS $ENV{VOLK_DIR}/lib
+        ${PC_VOLK_LIBDIR}
+    PATHS /usr/local/lib
+          /usr/local/lib64
+          /usr/lib
+          /usr/lib64
+          "@CMAKE_INSTALL_PREFIX@/lib"
+)
+
+INCLUDE(FindPackageHandleStandardArgs)
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(VOLK DEFAULT_MSG VOLK_LIBRARIES VOLK_INCLUDE_DIRS)
+MARK_AS_ADVANCED(VOLK_LIBRARIES VOLK_INCLUDE_DIRS)
-- 
2.1.4

